doc/src/manual/types.md

```diff
@@ -973,6 +966,39 @@ julia> Ptr{Int64} <: Ptr
 true
 ```
 
+## UnionAll Types
+
+We have said that a parametric type like `Ptr` acts as a supertype of all its instances
+(`Ptr{Int64}` etc.). How does this work? `Ptr` itself cannot be a normal data type, since without
+knowing the type of the referenced data the type clearly cannot be used for memory operations.
+The answer is that `Ptr` (or other parametric type like `Array`) is a different kind of type called a
+`UnionAll` type. Such a type expresses the *iterated union* of types for all values of some variable.
+
+`UnionAll` types are usually written using the keyword `where`. For example `Ptr` could be more
+accurately written as `Ptr{T} where T`, meaning all values whose type is `Ptr{T}` for some value
+of `T`. Each `where` introduces a single variable, so these expressions are nested for types with
```

Comment:
I know it's completely correct for mathematical literature, but earlier, our manual defines variable as "a name associated (or bound) to a value." And, afaict, we currently only refer to a variable in that programming sense. I suggested "Type parameter", since I think it might be sufficiently unambiguous to refer to both the variable and it's usage with the same term. I'm also fine with just consistently calling them `Type-variables`, such that it is always used in the manual as a compound term for disambiguation with variables.